SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -3078,6 +3089,26 @@ void CEulerSolver::SetInitialCondition(CGeometry **geometry, CSolver ***solver_c
                     (config->GetUnsteady_Simulation() == DT_STEPPING_2ND));
   bool SubsonicEngine = config->GetSubsonicEngine();
 
+  /*--- Check if a verification solution is to be computed. ---*/
+  if ((VerificationSolution)  && (ExtIter == 0) && !restart) {
+
+    /*--- Loop over the multigrid levels. ---*/
+    for (iMesh = 0; iMesh <= config->GetnMGLevels(); iMesh++) {
+      
+      /*--- Loop over all grid points. ---*/
+      for (iPoint = 0; iPoint < geometry[iMesh]->GetnPoint(); iPoint++) {
+        
+        /* Set the pointers to the coordinates and solution of this DOF. */
+        const su2double *coor = geometry[iMesh]->node[iPoint]->GetCoord();
+        su2double *solDOF     = solver_container[iMesh][FLOW_SOL]->node[iPoint]->GetSolution();
```

Comment:
From an idiomatic point of view it would be clearer if you put the manufactured solution into a tmp variable and then called SetSolution on the node (also good practice not to get pointers to private variables). Three calls to "getter" methods makes it look you are not doing anything.

Reply:
I did that to avoid an additional copy, but that just comes from my old habit that copies must be avoided, if possible. We can easily change that, if that improves readability.

Reply:
Yeah performance wise this is probably better, virtual methods are involved so I doubt the compiler could optimize away the copy. In any case in the call to the "Get" in the verification class maybe say it will write directly to the node.

Reply:
Performance is not really an issue here, because a) it is in the initialization phase and b) performance is of secondary, it not less, importance for the verification cases. Readability is much more important. So if it is clearer with the additional copy, we should do this.

Reply:
Should we change this to Get and Set?

Reply:
I would, but a comment saying "Set" on that last "Get" would do just as well

Reply:
It is now mentioned clearly in the comments.